src/serialize/flatten.js

```diff
@@ -0,0 +1,30 @@
+const { flatten } = require("flatley");
+const { cloneDeep } = require("lodash");
+
+const hasPeriodInKey = (key, _) => {
+  if (key.indexOf(".") >= 0) {
+    throw new Error("Key names must not have . in them");
+  }
+  return false;
+};
+
+const filters = [{ test: hasPeriodInKey }];
+
+/**
+ * Calls external flatten library but ensures that global filters are always applied
+ * @param {*} data
+ * @param {*} options
+ */
+const flattenWithGlobalFilters = (data, options) => {
+  const newOptions = cloneDeep(options) || {};
+  if (newOptions.coercion) {
+    newOptions.coercion.push(...filters);
+  } else {
+    newOptions.coercion = filters;
+  }
+  return flatten(data, newOptions);
```

Comment:
```javascript
return flatten(data, {
   ...options,
   coercion: [
     ...options.coercion,
     ...filters,
   ]
```
looks equivalent

Reply:
looks harder to read

Reply:
really ? I find it easier to understand than the if/else condition

```
const newOptions = cloneDeep(options) || {};
newOptions.coercions = (options.coercions || []).concat(filters);
return flatten(data, newOptions);
```

is this better ?

Reply:
the if else reads closer to english

if there's an existing coercion object, append to it
else set it to thing

just stylistic difference